pkg/cmd/step/git/credentialhelper/step_git_credentialhelper.go

```diff
@@ -0,0 +1,159 @@
+package credentialhelper
+
+import (
+	"os"
+
+	"github.com/jenkins-x/jx/pkg/auth"
+	"github.com/jenkins-x/jx/pkg/cmd/opts/step"
+	"github.com/jenkins-x/jx/pkg/gits/credentialhelper"
+	"github.com/pkg/errors"
+
+	"github.com/jenkins-x/jx/pkg/cmd/helper"
+
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/templates"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/spf13/cobra"
+)
+
+const (
+	optionGitHubAppOwner = "github-app-owner"
+)
+
+// StepGitCredentialHelperOptions contains the command line flags
+type StepGitCredentialHelperOptions struct {
+	step.StepOptions
+
+	GitKind        string
+	GitHubAppOwner string
+}
+
+var (
+	stepGitCredentialHelperLong = templates.LongDesc(`
+		This pipeline step generates a Git credentials file for the current Git provider secrets
+
+`)
+
+	stepGitCredentialHelperExample = templates.Examples(`
+		# respond to a git credentials request
+		jx step git credentials-helper
+`)
+)
+
+// NewCmdStepGitCredentialHelper creates a new command
+func NewCmdStepGitCredentialHelper(commonOpts *opts.CommonOptions) *cobra.Command {
+	options := StepGitCredentialHelperOptions{
+		StepOptions: step.StepOptions{
+			CommonOptions: commonOpts,
+		},
+	}
+	cmd := &cobra.Command{
```

Comment:
I think a dedicated command might not be a bad idea. Should probably be hidden in this case and we probably need to move some things around in order to not to duplicate code.